ouroboros-consensus/test-consensus/Test/Dynamic/General.hs

```diff
@@ -220,3 +262,53 @@ prop_general k TestConfig{numSlots, nodeJoinPlan, nodeTopology} schedule
       where
         NumSlots t = numSlots
         NodeJoinPlan m = nodeJoinPlan
+
+    -- check for Chain Growth violations if there are no Common Prefix
+    -- violations
+    --
+    -- We consider all possible non-empty intervals, so the interval span
+    -- @s@ varies but is always at least 1. We compute a different /speed
+    -- coefficient/ @τ@ for each interval under the assumption that there are
+    -- no message delays (ie @Δ = 0@). This is essentially a count of the
+    -- active slots for that interval in the refined @schedule'@.
+    prop_all_growth =
+        (.||.) (not isConsensusExcepected) $
+        conjoin $ zipWith f slotDepths (List.tails slotDepths)
+      where
+        f ::
+             (SlotNo, BlockNo, BlockNo)
+          -> [(SlotNo, BlockNo, BlockNo)]
+          -> Property
+        f (s1, _, max1) =
+            conjoin .
+            map (\(s2, min2, _) -> prop_growth (s1, max1) (s2, min2))
+
+        prop_growth :: (SlotNo, BlockNo) -> (SlotNo, BlockNo) -> Property
+        prop_growth (s1, b1) (s2, b2) =
+            counterexample (condense (s1, s2, b1, b2, numActiveSlots)) $
+            (.&&.)
+                (counterexample "negative chain growth" $
+                 property (b2 >= b1)) $
+            counterexample "insufficient chain growth" $
+            property (d >= toEnum numActiveSlots)
+          where
+            BlockNo d = b2 - b1
+            numActiveSlots =
+                Map.size $
+                flip Map.filterWithKey (getLeaderSchedule schedule') $
+                \slot ls -> s1 <= slot && slot < s2 && (not . null) ls
+
+        slotDepths =
+            [ case map snd bnos' of
+                  [] -> (slot, 0, 0)
+                  o  -> (slot, minimum o, maximum o)
+            | (slot, bnos) <- slotNodeDepths
+            , let bnos' = filter (joinedBefore slot . fst) bnos
+            ]
+
+        joinedBefore slot nid = nodeIdJoinSlot nodeJoinPlan nid < slot
+
+    slotNodeDepths =
```

Comment:
A short explanation of what the slot depth of a node is (and a type signature) would be welcome :slightly_smiling_face:

Reply:
Done; also renamed to `extrema`.